Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/DrawerController.java

```diff
@@ -143,31 +150,69 @@ public void onWatchCountClicked(Pin pin) {
     @Override
     public void onHeaderClicked(DrawerAdapter.HeaderAction headerAction) {
         if (headerAction == DrawerAdapter.HeaderAction.CLEAR || headerAction == DrawerAdapter.HeaderAction.CLEAR_ALL) {
-            boolean all = headerAction == DrawerAdapter.HeaderAction.CLEAR_ALL || !ChanSettings.watchEnabled.get();
-            final List<Pin> pins = watchManager.clearPins(all);
-            if (!pins.isEmpty()) {
+            final boolean all = headerAction == DrawerAdapter.HeaderAction.CLEAR_ALL || !ChanSettings.watchEnabled.get();
```

Comment:
Add a similar dialog for when a user attempts to delete a pin (one, not all of them) that has a download associated to it. Show the dialog only once so the user knows what it does.

Reply:
Nah.